app/api/schema/users.py

```diff
@@ -272,3 +275,19 @@ class Meta:
         type_='event',
         many=True,
     )
+
+    @pre_dump
+    def handle_deleted_users(self, data):
+        if (
+            not (
+                is_logged_in
+                and current_user
+                and data
+                and (current_user.is_staff or current_user.id == data.id)
+            )
+            and data.deleted_at != None
+        ):
+            data.email = 'deleted@eventyay.com'
+            data.first_name = 'deleted'
+            data.last_name = 'user'
```

Comment:
Please return a new user instance, not modify the data

Reply:
@iamareebjamal so should I copy data into the new object and modify the fields then? something like
`user = data.copy()
            user.email = 'deleted@eventyay.com'
            user.first_name = 'deleted'
            user.last_name = 'user'`

Reply:
Just create a new object of User model

Reply:
Ohh I finally understood what you meant by data leak previously we just have to return these attributes and not copy the other fields :sweat_smile: :facepalm:

Reply:
@iamareebjamal we aren't saving this instance to the db right? I am getting an error to set the id for this user since user model has function _is_role with line `uer = UER.query.filter_by(user=self, role=role).first()` which is trying to find user.id column